### PR TITLE
Add location service to LocationComponentManager and DroneExecutor

### PR DESCRIPTION
**Pull request by Jorque**
Adds the location service to LocationComponentManager to be able to zoom on the user location when he opens the map.
Adds the location service to DroneExecutor to allow the drone to fly back to the user.

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/drone3d/map/MapboxUtility.kt [hunk 1]
```diff
@@ -19,24 +19,31 @@ class MapboxUtility {
...
-         * Zoom on the first step of a  [missionPath] on the map [mapboxMap].
+         * Zoom on the first step of a [missionPath] on the map [mapboxMap].
          */
-        fun zoomOnMission(missionPath: List<LatLng>, mapboxMap: MapboxMap){
+        fun zoomOnMission(missionPath: List<LatLng>, mapboxMap: MapboxMap) {
 
-            if(missionPath.isNotEmpty()){
+            if (missionPath.isNotEmpty()) {
 
                 val firstCoordinates = LatLng(missionPath[0].latitude, missionPath[0].longitude)
```
Comment (Squalene) [inline app/src/main/java/ch/epfl/sdp/drone3d/map/MapboxUtility.kt:33]: As missionPath is already made up of LatLng, you could directly have  
``` val firstCoordinates = missionPath[0]```

File: app/src/main/java/ch/epfl/sdp/drone3d/service/impl/drone/DroneExecutorImpl.kt [hunk 2]
```diff
@@ -25,135 +26,175 @@ import kotlin.math.roundToInt
...
     }
 
     override fun returnToUserLocationAndLand(context: Context) {
-        // TODO Query user position
-        val returnLocation = Telemetry.Position(.0, .0, 10f, 10f)
-
-        val completable = getConnectedInstance() ?: throw IllegalStateException("Could not query drone instance")
-        val instance = service.provideDrone() ?: throw IllegalStateException("Could not query drone instance")
+        if (!locationService.isLocationEnabled()) {
+            // What TODO when cant retrieve user location ?
```
Comment (Squalene) [inline app/src/main/java/ch/epfl/sdp/drone3d/service/impl/drone/DroneExecutorImpl.kt:124]: Maybe the function should only be called after checking  ```if (locationService.isLocationEnabled())``` and throw IllegalStateException() if the function is called when location is disabled.
On top of that, a Toast could be thrown if the button is pressed but location service is disabled.

File: app/src/main/java/ch/epfl/sdp/drone3d/service/impl/drone/DroneExecutorImpl.kt [hunk 3]
```diff
@@ -25,135 +26,175 @@ import kotlin.math.roundToInt
...
+            return
+        }
+        val userPosition = locationService.getCurrentLocation()!!
+        val returnLocation =
+            Telemetry.Position(userPosition.latitude, userPosition.longitude, 10f, 10f)
```
Comment (Squalene) [inline app/src/main/java/ch/epfl/sdp/drone3d/service/impl/drone/DroneExecutorImpl.kt:129]: I think the relative altitude should be 0 in the telemetry position and the absolute altitude optional if possible.

File: app/src/main/java/ch/epfl/sdp/drone3d/service/impl/drone/DroneExecutorImpl.kt [hunk 4]
```diff
@@ -25,135 +26,175 @@ import kotlin.math.roundToInt
...
+
+        val completable =
+            getConnectedInstance() ?: throw IllegalStateException("Could not query drone instance")
+        val instance =
+            service.provideDrone() ?: throw IllegalStateException("Could not query drone instance")
 
         goToLocation(context, completable, instance, returnLocation)
     }
 
     private fun distance(pos: Telemetry.Position, returnLocation: Telemetry.Position): Int {
```
Comment (Squalene) [inline app/src/main/java/ch/epfl/sdp/drone3d/service/impl/drone/DroneExecutorImpl.kt:139]: Is there a reason for the distance to be forced to Int?

File: app/src/main/java/ch/epfl/sdp/drone3d/service/impl/drone/DroneExecutorImpl.kt [hunk 5]
```diff
@@ -25,135 +26,175 @@ import kotlin.math.roundToInt
...
         )
     }
+
+    private fun performLanding(
+        instance: System,
+        pos: Telemetry.Position,
+        returnLocation: Telemetry.Position
+    ) {
+        val isRightPos = distance(pos, returnLocation) == 0
+        val isStopped = data.getSpeed().value?.roundToInt() == 0
```
Comment (Squalene) [inline app/src/main/java/ch/epfl/sdp/drone3d/service/impl/drone/DroneExecutorImpl.kt:195]: Maybe veriying that the speed() is under a certain float treshold, rounding to Int seems arbitrary. 
If speed is in m/s, 0.5 meters per second might still be fast

**Review by Squalene — APPROVED** [review]
Expect rounding to Int that I did not understand, nice job

**Review by GabrielFleischer — APPROVED** [review]
Seems ok to me

**Pull request merged**